### ISSUE TITLE
Check if a budget is set

### ISSUE THREAD

**Issue by dstallenberg** [syntest-framework/syntest-core#269]
Check if atleast one of the budget's is not equal to MAX_SAFE_INTEGER

**Comment by mitchellolsthoorn** [comment]
Currently, all search budgets managed by the BudgetManager have the default set to the max number available for that type:

https://github.com/syntest-framework/syntest-core/blob/88813984d20845cfac1b1cd3a1fdbaef5f23d8dc/tools/base-language/lib/Configuration.ts#L278-L313

This means that if the user doesn't specify a search budget the framework will keep searching until it has covered all objectives. This default search budget (i.e., objective manager) is always active. We could also decide that this could be the default behavior.

Since getting 100% percent coverage in many cases is not feasible, it may create undesirable situations. However, setting a default search budget based on time or the number of evaluations seems undesirable as well. So, the objective manager could act as a natural search budget that is always active. We could post a warning on the CLI/UI if no other search budget was active (i.e., changed from the default). This could either be just a colored warning or a question asking if the user wants to continue (with an always yes overwrite -y flag)

According to https://yargs.js.org/docs/#api-reference-numberkey a number can also be empty with will result in the value of undefined. I am not sure if this can cause problems with our budget manager, since it expects a value. The budgets are internally also used for the metrics collection so they should be defined. We can decide to always instantiate the budgets and either provide the max value if undefined or make the budgets aware if they are active or not, by checking if they are changed from their max value, this might make the metrics cleaner as well.

Or do we still want to check if one of them is defined?

To summarize:
- [ ] Keep the objective manager (i.e., stop when all objectives are covered) as the only default-enabled search budget
- [ ] Post a warning to the user if no other budget was enabled (e.g., through a colored warning or a prompt)
- [ ] Instantiate the budget manager with the max value if the value undefined was specified (not sure this is necessary)
- [ ] Make the budgets aware if they are enabled
- [ ] Only save the max budgets to the metric manager that are enabled (We could also decide that we want to keep these in the metric manager just to have them around)

@dstallenberg What do you think?
